### PR TITLE
mgmt, NetworkAnalytics, add live tests

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#35316](https://github.com/Azure/azure-sdk-for-java/issues/35316)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import com.azure.resourcemanager.resources.ResourceManager;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.util.Random;
+
+public class NetworkAnalyticsManagerTests extends TestBase {
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.create("centraluseuap", "Central US EUAP");
```
Comment (weidongxu-microsoft) [inline sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:30]: why EUAP?

can it work in usually e.g. westus?

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    @Disabled("Missing required property: workload. Paths in payload: '$.properties.workload'")
```
Comment (weidongxu-microsoft) [inline sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: what's this?

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:30]: image
image
The only locations that support `Microsoft.NetworkAnalytics/DataProducts` in the subscription I can use are `Central US EUAP` and `East US 2 EUAP`.

Comment (weidongxu-microsoft) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:30]: What about other resources?

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java (hunk 2 above)
Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: The `Workload` is a required item while create data product on portal
image
but there is no such setting interface in model [`DataProductProperties`](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/main/java/com/azure/resourcemanager/networkanalytics/models/DataProductProperties.java),
so disabled the live test case.

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:30]: https://learn.microsoft.com/en-us/azure/operator-insights/data-product-create?tabs=azure-cli

Comment (weidongxu-microsoft) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:30]: None of it says EUAP?

But major problem is below the `@Disabled`
https://github.com/Azure/azure-sdk-for-java/pull/40402#discussion_r1620087502

If we disable the only test in live test, what we are running it for?

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: Can you create it e.g. via CLI?

Use the `--debug` option to print out the PUT request.
If you can create it via CLI, and it provide the "workload", while we don't see it in Swagger, we need to report it to service ASAP.

Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: Can not create by CLI, and CLI does not support it.
https://learn.microsoft.com/en-us/cli/azure/network-analytics/data-product?view=azure-cli-latest#az-network-analytics-data-product-create

Comment (weidongxu-microsoft) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: Then create it via Portal

Comment (weidongxu-microsoft) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: > Can not create by CLI, and CLI does not support it. https://learn.microsoft.com/cli/azure/network-analytics/data-product?view=azure-cli-latest#az-network-analytics-data-product-create

Where in that page say CLI does not support it?

Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: image
The parameter "workload" is not found in the parameter list of the command `az network-analytics data-product create`, and it does not use the following abnormalities will be caused:
image

Comment (weidongxu-microsoft) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: So CLI runs and create the resource. But the resource is not correct? Please report it at https://github.com/Azure/sdk-release-request/issues/4914

How about create via Portal?

**Comment by weidongxu-microsoft** [comment]
@v-hongli1 
And here it test in .NET
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/networkanalytics/Azure.ResourceManager.NetworkAnalytics/tests/NetworkAnalyticsManagementTestBase.cs#L45-L55

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java (hunk 2 above)
Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: Regardless of whether you create through the portal or the CLI, you must apply for access rights for the subscription.
https://learn.microsoft.com/en-us/azure/operator-insights/data-product-create?tabs=azure-cli#prerequisites

Comment (weidongxu-microsoft) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:74]: Ok, then comment out the CRUD part (and no `@embedStart` `@embedEnd`).

Just leave a List, if it works. We at least need to call some API in a LIVE test.

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java [hunk 3]
```diff
@@ -40,63 +23,15 @@ public void beforeTest() {
...
-            if (dataProduct != null) {
-                networkAnalyticsManager.dataProducts().deleteById(dataProduct.id());
-            }
-        }
-    }
-
-    private static String randomPadding() {
-        return String.format("%05d", Math.abs(RANDOM.nextInt() % 100000));
+    public void testListDataProducts() {
+        // @embedStart
```
Comment (weidongxu-microsoft) [inline sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:33]: do not add this.

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java [hunk 4]
```diff
@@ -40,63 +23,15 @@ public void beforeTest() {
 
         networkAnalyticsManager = NetworkAnalyticsManager
             .configure()
-            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
```
Comment (weidongxu-microsoft) [inline sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:26]: use BASIC

Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java]: Fixed in the new version.

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        networkAnalyticsManager = NetworkAnalyticsManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
```
Comment (weidongxu-microsoft) [inline sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java:31]: use `@LiveOnly` -- this is newly added in azure-core-test

**Review by weidongxu-microsoft — COMMENTED** [review]
Give me a pipeline run of this mgmt - test

File: sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java (hunk 5 above)
Comment (v-hongli1) [reply sdk/networkanalytics/azure-resourcemanager-networkanalytics/src/test/java/com/azure/resourcemanager/networkanalytics/NetworkAnalyticsManagerTests.java]: Fixed in the new version.

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**